src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -7259,15 +7264,21 @@ namespace SurfaceGeometry {
 
             Roughness = state.dataIPShortCut->cAlphaArgs(3);
             // Select the correct Number for the associated ascii name for the roughness type
-            if (UtilityRoutines::SameString(Roughness, "VeryRough")) state.dataSurface->ExtVentedCavity(Item).BaffleRoughness = VeryRough;
-            if (UtilityRoutines::SameString(Roughness, "Rough")) state.dataSurface->ExtVentedCavity(Item).BaffleRoughness = Rough;
-            if (UtilityRoutines::SameString(Roughness, "MediumRough")) state.dataSurface->ExtVentedCavity(Item).BaffleRoughness = MediumRough;
-            if (UtilityRoutines::SameString(Roughness, "MediumSmooth")) state.dataSurface->ExtVentedCavity(Item).BaffleRoughness = MediumSmooth;
-            if (UtilityRoutines::SameString(Roughness, "Smooth")) state.dataSurface->ExtVentedCavity(Item).BaffleRoughness = Smooth;
-            if (UtilityRoutines::SameString(Roughness, "VerySmooth")) state.dataSurface->ExtVentedCavity(Item).BaffleRoughness = VerySmooth;
+            if (UtilityRoutines::SameString(Roughness, "VeryRough"))
```

Comment:
This if-ladder (which should really be an if-else-if ladder, but that's besides the point) should really be a call to a function that loops over an array of `SurfaceRoughness` names and returns the index to it, which you can then static_cast to a `SurfaceRoughness`. 

Something like: 

`constexpr std::array<std::string_view, SurfaceRoughness::Num> SurfaceRoughnessNames = {"Very Rough", "Rough", ... }; // in the .hh file
`
`BaffleRoughness = static_cast<SurfaceRoughness>FindEnumeratedValueIndex(Roughness, SurfaceRoughnessNames);`

The presence of gsl::span should allows us to create a FindEnumeratedValueIndex that takes different std::array's as arguments.

Reply:
I'll do this on the next commit. Thanks!